apps/api/src/server/repositories/document.repository.js

```diff
@@ -58,6 +58,34 @@ export const getByCaseId = ({ caseId, skipValue, pageSize }) => {
 	});
 };
 
+/**
+ * Get a paginated array of documents by caseid ignoring migration cases (this is used for creating a new document reference so essential we ignore migrated
+ * references since they don't follow the same pattern)
+ *
+ * @param {{ caseId: number, skipValue?: number, pageSize?: number }} _
+ * @returns {import('@prisma/client').PrismaPromise<Document[]>}
+ */
+export const getByCaseIdExcludingMigrationDocumentReferences = ({
```

Comment:
I am wondering whether we need this new fn to handle paged data - before we were re-using an existing fn which is also used elsewhere.  Now we have our own fn to get this value, do we actually want to just get the doc with the latest doc reference, and return the number part of that?

Reply:
That's a really good point, I just duplicated the existing one and planned to clean up. Thanks, will make the change!